### PR TITLE
Cmake xcode generator project fix

### PR DESCRIPTION
**Pull request by ghost**
It is now possible to select Xcode as the generator. For the project to build the SDLMain source and header file had to be included.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -242,9 +242,16 @@ if (NOT THREE_HEADER_ONLY)
   if (NOT THREE_LIBRARY_STATIC)
     add_definitions(-DTHREE_DYN_LINK)
   endif()
+  
+  set(THREE_LIB_SOURCE_FILES three/impl/src.cpp three/impl/src_extras.cpp)
+
+  if(${CMAKE_GENERATOR} STREQUAL Xcode)
+    set(THREE_LIB_SOURCE_FILES ${THREE_LIB_SOURCE_FILES} externals/sdl-1.2.15/xcode/SDLMain.m)
```
Comment (jdduke) [inline CMakeLists.txt]: This looks great, thanks!  What if inside this branch we instead have something like:

add_library(xcode_sdlmain STATIC externals/sdl-1.2.15/xcode/SDLMain.m)
set(SDLMAIN_LIBRARY xcode_sdlmain)

Then ${SDLMAIN_LIBRARY} will get picked up automatically for the examples, and we're no longer include the SDLMain source file in the library itself.  What do you think?

**Comment by ghost** [comment]
I tried using the static library, but kept getting the warning "Category is implementing a method which will also be implemented by its primary class", which was due to the method signature:
https://github.com/elwinarens/SDLMain.m/commit/615eb30cbb617b3127030f807e65b7165e53cd11 (error = warning :-))

And because of the -Werror flag it wouldn't build. I guess the warning could be suppressed.  What would you prefer?

**Comment by ghost** [comment]
Maybe i was fixing the symptoms.. ?

**Comment by jdduke** [comment]
I'm not familiar with this warning, so it's hard to say how dangerous it is.  I think either the fix to which you linked, or a warning suppression that wraps the terminate function would be fine, e.g.:

#pragma clang diagnostic push
#pragma clang diagnostic ignored “-Wobjc-protocol-method-implementation”
@implementation SDLApplication
/\* Invoked from the Quit menu item */
- (void)terminate:(id)sender
  ...
  @end
  #pragma clang diagnostic pop

(as long as it runs with one of those solutions, I'll be happy :)

**Comment by ghost** [comment]
I've updated the cmakelists as you suggested. Examples successfully build :+1:

**Comment by jdduke** [comment]
Excellent, is the warning still giving you problems? Or has that been resolved?

**Comment by ghost** [comment]
This builds all projects without the warning. The warning is already fixed in the included SDLMain.m file.

**Comment by jdduke** [comment]
Thanks!

**Comment by ghost** [comment]
Thank you too :)

**Pull request merged**